packages/hyperion-autologging/src/ALElementInfo.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+ */
+
+'use strict';
+
+import { ComponentNameValidator, defaultComponentNameValidator, getReactComponentData_THIS_CAN_BREAK } from './ALReactUtils';
+import type { ReactComponentData } from './ALReactUtils';
+import { getVirtualPropertyValue, setVirtualPropertyValue } from '@hyperion/hyperion-core/src/intercept';
+
+const AL_ELEMENT_INFO_PROPNAME = '__alInfo';
+const AUTO_LOGGING_COMPONENT_TYPE = 'data-auto-logging-component-type';
+
+let componentNameValidator: ComponentNameValidator = defaultComponentNameValidator;
+/**
+ * Set a component validator function to use when extracting
+ * a component name for the event, utilized in ALElementInfo.
+ * The stack will remain unfiltered, but component name linked must be valid via validator.
+ * @param validator: callable passed a component name returning true if valid
+ */
+export function setComponentNameValidator(validator: ComponentNameValidator): void {
```

Comment:
technically, this could have been added to the `ALReactUtils` itself for `sedDefaultComponentNameValidator` so that any other module in the future can simply call the corresponding api without worrying about adding this stuff. 

I'll merge this commit, but you can fix it later.

Reply:
Makes sense.  I did just realize though,  that with this static approach whoever calls the API second (surface mutation publisher in current test app) will overwrite previous caller config (ui event publisher in test app).  I don't see a use case for the validators to be different,  but might be something to account for with shared init options so we only allow one value in the config, which gets passed down to modules that require them.